config/routes.rb

```diff
@@ -93,13 +93,16 @@
         end
       end
     end
-    resources :teams, except: [:edit, :update, :destroy] do
+    resources :teamsets, only: [:index, :edit, :update] do
       member do
-        patch :dissolve
-      end
-      collection do
-        patch :dissolve_all
+        patch :clone
         patch :randomize
+        patch :dissolve_all
+      end
+      resources :teams, only: [:show] do
```

Comment:
And as for inconsistent intermediate ids, see above: those are useful auditable problems that help both catch bugs in our code and attribute potential hacking attempts to their culprits...